### PR TITLE
feat(routing): add ensemble response synthesis

### PR DESCRIPTION
**Pull request by pst2154**
## What

Adds an `ensemble` route: two to four targets generate drafts concurrently, then a configured synthesizer returns one answer. Supports independent candidate token budgets, a minimum usable-candidate threshold, and a synthesis prompt override. Internal drafts are buffered and reasoning blocks excluded; the final response retains the caller's streaming and output settings.

## Why

Allows applications to explore whether multiple smaller models can produce useful answers at lower latency than a stronger single model. The included deployment compares SOL + Opus → SOL against Astra, with both candidates required to contribute.

## Benchmarks

Exploratory live evaluation on September 9, 2026 through an authenticated OpenAI-compatible gateway (endpoint omitted):

- Baseline: `azure/openai/gpt-6-astra`.
- Candidates: `azure/openai/gpt-5.6-sol` and `azure/anthropic/claude-opus-5`; synthesizer: SOL. Three upstream calls per ensemble request.
- Budgets: 2,048 output tokens per candidate; 1,024 for the final answer. Both candidates must return usable output.
- Six manually selected tasks. Baseline and ensemble requests started together in the initial run.
- Judge: `azure/openai/gpt-5.6-terra`, with anonymous A/B answers and alternating order. Scores are correctness, completeness, instruction following, and clarity, each out of five.

| Case | SOL + Opus → SOL | Astra | Judge score, fusion / Astra |
|---|---:|---:|---:|
| Tokio first-success review | 33.27 s | 125.30 s | 20 / 20 |
| Python LRU debugging | 12.46 s | 16.49 s | 20 / 20 |
| Two-test Bayesian posterior | 10.66 s | 10.27 s | 20 / 20 |
| Boolean truth-value derivation | 10.33 s | 19.85 s | 20 / 20 |
| Exact JSON sorting | 6.10 s | 3.72 s | 20 / 20 |
| Transactional outbox, original run | 35.62 s | 46.92 s | 20 / 20 |
| Transactional outbox, fusion-only rerun | 28.13 s | — | 20 / 20 against saved Astra answer |

Fusion was faster on four of six initial cases; Astra was faster on Bayes and JSON sorting. The committed report's mean of 16.83 s versus 37.09 s and median of 11.56 s versus 18.17 s substitute the outbox rerun for its original result, so they combine two prompt versions and should not be treated as a controlled benchmark.

All six pairs received ties from the single model judge, but these small tasks, development-time prompt tuning limit the conclusion. This does not establish general quality superiority or equivalence. There are no repeated samples, confidence intervals, or independent held-out evaluation. Returned terminal usage excludes candidate usage, so no cost advantage is claimed.

### Evidence and configuration

- [Committed benchmark report](https://github.com/pst2154/Switchyard/blob/1cf7dcfa/benchmark/ensemble-inference-api-evidence.md)
- [Secret-free deployment configuration](https://github.com/pst2154/Switchyard/blob/1cf7dcfa/benchmark/server-configs/inference-api-ensemble-sol-opus-vs-astra.toml)
- [Ensemble configuration and behavior documentation](https://github.com/pst2154/Switchyard/blob/1cf7dcfa/docs/routing_algorithms/ensemble_routing.md)

## Validation

- Full Rust workspace tests passed after rebasing onto upstream `main`.
- Eight ensemble unit tests cover concurrent calls, candidate failure, minimum usable results, token budgets, reasoning removal, and configuration validation.
- Four runner configuration tests cover construction and invalid settings.
- Public HTTP integration test verifies both candidate calls, both drafts reaching synthesis, separate candidate/final budgets, and the returned synthesized answer.
- Earlier Clippy, Ruff, mypy, strict docs build, and 115 Python tests passed. The Docker integration file containing two tests was excluded because the Docker daemon was unavailable; the full Python suite was not green.

## Notes for reviewers

Start with `crates/libsy/src/algorithms/ensemble.rs`, then the runner configuration and HTTP integration test. Candidate failures can be tolerated up to the configured threshold; all-candidate failures retain a typed error. Candidate generation waits for all candidates before synthesis, and the final response usage does not aggregate internal call costs.

### CODE REVIEW

File: docs/routing_algorithms/ensemble_routing.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# Ensemble Routing
+
+Ensemble routing calls two to four candidate targets concurrently, buffers the
+successful responses, and asks a synthesizer target to produce one final answer.
+It is response-level fusion: Switchyard does not merge model weights or logits.
+
+Use it when answer quality can justify several candidate calls plus synthesis.
+Compared with ordinary routing, every request costs at least three model calls
```
[the remark this answers is not in the document]
Comment (pst2154) [reply docs/routing_algorithms/ensemble_routing.md]: Codex replying: fixed in 6d6f7e6b. The three-call minimum now applies only to requests that reach synthesis. The docs explicitly state that falling below minimum_successful_candidates returns an error after the candidate calls, without calling the synthesizer. Verified against the current threshold check and passed the strict MkDocs build (cd docs && make publish) and git diff --check.
